src/Knp/Menu/MenuItem.php

```diff
@@ -477,22 +485,22 @@ public function isCurrent(): ?bool
 
     public function isLast(): bool
     {
-        // if this is root, then return false
-        if ($this->isRoot()) {
+        // if this is root or there is no parent, then return false
+        if ($this->isRoot() || null === $parent = $this->getParent()) {
```

Comment:
I would change that to `if (null === $this->parent)` (i.e. explicitly doing the condition done in `isRoot`) and then using `$this->parent->getLastChild()` below.

Being the root is exactly not having a parent